### ISSUE TITLE
Consider moving header and DA header into proposal

### ISSUE THREAD

**Issue by adlerjohn** [lazyledger/lazyledger-specs#163]
Ref #162.

**Comment by liamsi** [comment]
To settle the core data types once and for all and to actually specify everything we need for the MVP (here MVP = nodetypes beyond light clients), the next PR on the Consensus messages should actually specify the following:

If we decide to keep the DAHeader outside of the Proposal, how are validators expect to receive it? (bare minimum: via broadcast (push) or via requesting (pull)). If it is broadcast, I want to see a single reason _why_ we do not include it in the Proposal Message directly (as it is broadcasted anyways).

For each field that gets added to the Proposal (or other consensus messages), the spec MUST specify what to do with these fields (for the different node types). Even for the DAHeader this holds (at this point it probably clear to everyone what to do with the DAHeader but what if we hired another engineer and they turn to the spec for answers?).

Here is what I think (currently): the proposal should contain _all_ data that is necessary to validate the proposal but it _should not_ contain a bunch of redundant fields (unless they really add immense value). 
If that means adding the fields that were removed in #162, the spec must specify how these fields are supposed to be used.
Otherwise, engineers have to either guess ask you. if you know the answer, why not write it down and save time answering as well as discussing with me.
